vector/src/main/java/im/vector/app/features/login/qr/QrCodeLoginViewModel.kt

```diff
@@ -58,20 +71,60 @@ class QrCodeLoginViewModel @AssistedInject constructor(
     }
 
     private fun handleOnQrCodeScanned(action: QrCodeLoginAction.OnQrCodeScanned) {
-        if (isValidQrCode(action.qrCode)) {
-            setState {
-                copy(
-                        connectionStatus = QrCodeLoginConnectionStatus.ConnectingToDevice
-                )
+        Timber.tag(TAG).d("Scanned code: ${action.qrCode}")
+
+        val rendezvous = try { Rendezvous.buildChannelFromCode(action.qrCode) } catch (t: Throwable) {
+            Timber.tag(TAG).e(t, "Error occurred during sign in")
+            if (t is RendezvousError) {
+                onFailed(t.reason)
+            } else {
+                onFailed(RendezvousFailureReason.Unknown)
+            }
+            return
+        }
+
+        setState {
+            copy(
+                    connectionStatus = QrCodeLoginConnectionStatus.ConnectingToDevice
+            )
+        }
+
+        _viewEvents.post(QrCodeLoginViewEvents.NavigateToStatusScreen)
+
+        viewModelScope.launch(Dispatchers.IO) {
```

Comment:
We should not use `Dispatchers` directly as it is not testable. We should instead use the `coroutineDispatchers` of `Session`. Best version would be to switch internally in the SDK to IO dispatcher so that client does not have to worry about that.

Reply:
Please can we refactor this when we adde the tests rather than blocking this PR?

Reply:
Sure, this a non blocking point for now.

Reply:
I think it's just calling session.coroutineDispatchers.io

Reply:
I was thinking of the "best version" which requires some refactoring in the SDK to make sure that the right context is passed around.